### ISSUE TITLE
Error 'Permissions needed: dashboards:read' when switching from 8.X to 9.X for anonymous access

### ISSUE THREAD

**Issue by lukaszgryglicki** [grafana/grafana#54974]
**What happened**:

I was trying to update Grafana to newest for [CNCF DevStats](https://cncf.teststats.cncf.io) project (8.5.11 -> 9.1.3) and:
- This is [the version](https://envoy.devstats.cncf.io) that still uses 8.5.11 - everything work OK.
- This is [the version](https://cncf.teststats.cncf.io) where I'm attempting to use 9.1.3 - it fails for every single dashboard with:
```
Error
You'll need additional permissions to perform this action. Permissions needed: dashboards:read
```
Screenshot:
![Screenshot 2022-09-09 at 14 30 44](https://user-images.githubusercontent.com/2469783/189350604-6a1453d1-c4ef-4a8b-8b3f-b050d7e4d791.png)

I didn't change any user permissions, so this is strange. I can still sing-in as admin and then all works OK, so this is some kind of permission issue for anonymous user - everybody who visits DevStats page is NOT signed in and is NOT using any user account etc - they are just "viewers" by default and also have "viewers can edit" permission, so they can edit dashboards, execute queries etc. - they just acannot save which is exactly what I want. I use the following `grafana.ini` options for this:
```
auto_assign_org_role = Viewer
viewers_can_edit = true
(...)
[auth.anonymous]
enabled = true
```

- grafana.ini file is [here](https://github.com/cncf/devstats/blob/master/grafana/shared/grafana.ini.example#L210) - it is tweaked on the fly to replace all instance variables in `{{` and `}}` but this is just to personalize grafana instance for a given CNCF project.

**What you expected to happen**:

- I expected to be able to see all dashboards just like in 8.X.

**How to reproduce it (as minimally and precisely as possible)**:

- Go to [this URL](https://cncf.teststats.cncf.io/d/8/dashboards?orgId=1).

**Anything else we need to know?**:

**Environment**: Ubuntu Linux 22.04 LTS.
- Grafana version: 9.1.3 (while trying to port from 8.5.11).
- Data source type & version: I don't think it matters but postgres.
- OS Grafana is installed on: Ubuntu Linux 22.04 LTS.
- User OS & Browser: Ubuntu Linux 22.04 LTS Chrome.
- Grafana plugins: doesn't matter.
- Others: none.

**Comment by Razikus** [comment]
Same problem as mine
https://github.com/grafana/grafana/issues/54868

Did you found any workaround?

**Comment by Razikus** [comment]
Downgrading to 9.0.8 resolves that probably

**Comment by lukaszgryglicki** [comment]
Well I don't need to switch right now, I will rather wait for newest version that fixes this.
Thanks @Razikus

**Comment by lukaszgryglicki** [comment]
BTW: are there plans to fix it in newer versions of Grafana.

**Comment by kalleep** [comment]
Hey. Thanks for reporting this :)

I may have a suspicion on what the issue is but I need a little more info to confirm if it is the same problem.

1. Are you using provisioning to create your dashboards?
2. If you look at the permissions for one dashboard is only admin listed there?

**Comment by lukaszgryglicki** [comment]
1. I'm not using provisioning.
2. I'm not adding any permissions - all those dashboards are public - should be available for all users - there is no sign in or anything like that in this project.
3. When I log in as admin (this is the only user who can edit them) I can see permissions only for admin - but I didn't add/change them
<img width="2233" alt="Zrzut ekranu 2022-09-19 o 07 33 49" src="https://user-images.githubusercontent.com/2469783/190955085-7202646a-3140-4b24-8486-6fcd364fd144.png">
I cannot add permissions for the anonymous user because there is no such user.
Also, there are 40 or 50 dashboards and over 140 pages (for each CNCF project) so if I need to add something n for each dashboard I would need to do this via some kind of script - ideally some garafana.ini entry...

**Comment by vtorosyan** [comment]
@lukaszgryglicki thanks for getting back with more details. We are still investigating the issue, and have a hunch where the root cause is, however can't verify completely. With Grafana 9+ we have turned on RBAC (role-based access control) as an underlying system for access control in Grafana. When RBAC is being turned on, there are data migrations happening and we suspect that you are either facing a bug which we had or something went wrong with the migrations. 

To verify this assumption, I can share an SQL script which will reset all migrations and with the next start of Grafana all relevant migrations will run again. This of course if you are willing to run the script, otherwise we are looking mitigating this properly. 

Here are the steps you can follow:

1. Stop Grafana 
1. In your database, run the following queries:
```
DELETE
FROM builtin_role
where role_id IN (SELECT id
                  FROM role
                  WHERE name LIKE 'managed:%');

DELETE
FROM team_role
where role_id IN (SELECT id
                  FROM role
                  WHERE name LIKE 'managed:%');

DELETE
FROM user_role
where role_id IN (SELECT id
                  FROM role
                  WHERE name LIKE 'managed:%');

DELETE
FROM permission
where role_id IN (SELECT id
                  FROM role
                  WHERE name LIKE 'managed:%');

DELETE
FROM role
WHERE name LIKE 'managed:%';

DELETE
FROM migration_log
WHERE migration_id IN ('teams permissions migration',
                       'dashboard permissions',
                       'dashboard permissions uid scopes',
                       'data source permissions',
                       'data source uid permissions',
                       'managed permissions migration',
                       'managed folder permissions alert actions repeated migration',
                       'managed permissions migration enterprise');
```
      - These queries will reset the migration log and delete all so called "managed" permissions, which are a result of a translation of the old access control permissions to the new access control permissions. We have used these queries several times already and they are safe to run.
      - Above said, I'd recommend to keep a backup of the database as a best practice, so that you can recover if anything goes wrong.
3. Start Grafana again

**Comment by lukaszgryglicki** [comment]
Hi, don't have time for it, working 12+ hours everyday, I'll just wait when this is resolved and switch to `9.X` only then.
Also Grafana runs in K8s so I would have to exec bash inbto container then install sqlite there to do queries - no time blocks me.

**Comment by livegrenier** [comment]
Hi, I had the same problem and the SQL query suggested fixed it, thanks @vtorosyan

**Comment by wyhysj** [comment]
Same issue here in 9.2+, 9.1.8 is okay, brand new instance with google oauth provider, first login will show the error

**Comment by lukaszgryglicki** [comment]
Any news here? Is there a 9.X version that fixes this?

**Comment by lukaszgryglicki** [comment]
Any updates?

**Comment by vtorosyan** [comment]
Hi @lukaszgryglicki, sorry about late response. Have resetting the migrations as described [here](https://github.com/grafana/grafana/issues/54974#issuecomment-1252179049) helped? You would need to run that SQL only once and after that everything should be stable. 

If the SQL above does not fix the issue - I'd recommend to bump to the latest minor (if possible), as we had many other fixes which might be the root cause of what you experience.

We are also [updating](https://github.com/grafana/grafana/pull/59223/files) 9.0 upgrade instructions to mention that going back and forth between versions may introduce issues with permissions (that said, I am still not sure if what you see is caused by the upgrade/downgrade)

**Comment by lukaszgryglicki** [comment]
As I said, I can run it inside a grafana pod, but it will be lost when pod gets recreated. I can add this to pod startup script and see if that helps. But I will try newest 9.X first, I'll LYK.

**Comment by lukaszgryglicki** [comment]
So updating to the newest `9.1.3` is `NOT` solving the issue:
<img width="2359" alt="Zrzut ekranu 2022-12-2 o 12 01 17" src="https://user-images.githubusercontent.com/2469783/205278310-5a36ed06-97e5-4fc7-9c8e-32077d310acd.png">

Trying SQL approach...

**Comment by lukaszgryglicki** [comment]
OK I've `bash`'ed to Grafana pod running 9.1.3 and did:
```
vim sql.sql
cat sql.sql
DELETE FROM builtin_role where role_id IN (SELECT id FROM role WHERE name LIKE 'managed:%');
DELETE FROM team_role where role_id IN (SELECT id FROM role WHERE name LIKE 'managed:%');
DELETE FROM user_role where role_id IN (SELECT id FROM role WHERE name LIKE 'managed:%');
DELETE FROM permission where role_id IN (SELECT id FROM role WHERE name LIKE 'managed:%');
DELETE FROM role WHERE name LIKE 'managed:%';
DELETE FROM migration_log WHERE migration_id IN (
  'teams permissions migration',
  'dashboard permissions',
  'dashboard permissions uid scopes',
  'data source permissions',
  'data source uid permissions',
  'managed permissions migration',
  'managed folder permissions alert actions repeated migration',
  'managed permissions migration enterprise'
);
sqlite3 -echo -header -csv /var/lib/grafana/grafana.db
```

Then I attempted to open that Grafana instance but still exactly the same error.

Then I did a bunch of selects on Grafana database (inside the pod):
```
root@devstats-grafana-cncf-687858bf6f-lgh4r:/# sqlite3 -echo -header -csv /var/lib/grafana/grafana.db           
SQLite version 3.37.2 2022-01-06 13:25:41
Enter ".help" for usage hints.
sqlite> DELETE FROM builtin_role where role_id IN (SELECT id FROM role WHERE name LIKE 'managed:%');
DELETE FROM builtin_role where role_id IN (SELECT id FROM role WHERE name LIKE 'managed:%');
sqlite> select * from builtin_role;
select * from builtin_role;
sqlite> select * from team_role;
select * from team_role;
sqlite> \d
DATABASE                 DEFERRED                 DETACH                   DROP                     dashboard_encrypted      dashboard_public         dashboard_uid            data_keys                delete_key               display_name             
DEFAULT                  DELETE                   DISTINCT                 dashboard                dashboard_id             dashboard_snapshot       dashboard_version        data_source              description              
DEFERRABLE               DESC                     DO                       dashboard_acl            dashboard_provisioning   dashboard_tag            data                     datasource_uid           disable_resolve_message  
sqlite> select * from user_role;
select * from user_role;
sqlite> select * from permission;
select * from permission;
sqlite> select * from role;
select * from role;
sqlite> select * from migration_log;
select * from migration_log;
id,migration_id,sql,success,error,timestamp
(...) ---> a LOT of rows
sqlite> 
```

So SQL approach is not working, any other suggestions?

**Comment by lukaszgryglicki** [comment]
Reverted back to 8.5.15 (but kept that SQL in my init scripts, it seems to make no harm on 8.X and doesn't help in 9.X), still cannot use 9.x :(

**Comment by lukaszgryglicki** [comment]
cc @vtorosyan

**Comment by vtorosyan** [comment]
@lukaszgryglicki can you please share how is the provisioning of dashboards and folders are done? If the provisioning is not done through the interfaces that Grafana provides (e.g. API/provisioning files/terraform/UI), then the permissions that new access control system generates won't be generated. For example, if the provisioning is done by manual data ingestion, the permissions system won't work.

**Comment by lukaszgryglicki** [comment]
It is done manually, so the question is how can I fix permissions after they are done manually: 
- https://github.com/cncf/devstats/blob/master/grafana/shared/grafana_start.sh
- https://github.com/cncf/devstats/blob/master/grafana/shared/update_sqlite.sql

**Comment by lukaszgryglicki** [comment]
Any updates? This issue is open for 3 months and 5 days already and I provided all info that I was asked for and also tried suggested solution and its not working?
Any hope on this?

**Comment by vtorosyan** [comment]
@lukaszgryglicki thank you for the patience. At the moment I am afraid we don't have a good strategy to provide or promise any solution when the data is manipulated manually. The access control system has grown with it's complexity quite a bit with the 9.x release, and we are aiming for standardized experience of provisioning as supporting all possible scenarios is not feasible anymore.
 
I understand this is a show stopper for upgrade, sorry about that. 

Our recommendation would be to use one of the supported methods of provisioning (e.g. one of the following [Grafana Terraform](https://registry.terraform.io/providers/grafana/grafana/latest/docs/resources/dashboard), [Grafana Provisioning](https://grafana.com/docs/grafana/latest/administration/provisioning/#dashboards), UI or the [Dashboard API](https://grafana.com/docs/grafana/v8.5/http_api/dashboard/)).

**Comment by lukaszgryglicki** [comment]
Hi, I can't, I have a custom deployment.
Is there a way of fixing this via some SQL script? Myabe just another one?
Not deleting but maybe adding some default permissions/entries in tables(s)?

**Comment by lukaszgryglicki** [comment]
Ping?

**Comment by vtorosyan** [comment]
Hey, I'm afraid supporting custom SQL script or hacking around the way is not something we could commit to do and support long term, as the underlying complexity is not trivial to handle outside of the system boundaries. 

I understand that effort of using the supported methods is high, however that would bring stability long term and we could support better on our side.

**Comment by lukaszgryglicki** [comment]
i don't want custom SQL or anything like this - I just want to know how to enable any user to access dashboards - just the way it worked in 8.x.

**Comment by vtorosyan** [comment]
@lukaszgryglicki Unfortunately with the OSS and without using available public interfaces for provisioning resources, I can't offer any workaround that would make it work with the Grafana >9.

**Comment by GreenRover** [comment]
I face the same issue with grafana 10.2.0

Dashboards are imported via:
```
apiVersion: 1

providers:
- name: 'ServiceDescription'
  orgId: 1
  folder: ''
  type: file
  disableDeletion: false
  updateIntervalSeconds: 60 #how often Grafana will scan for changed dashboards
  options:
    path: /var/lib/my-dashboards
    foldersFromFilesStructure: true
```

My config (grafana.ini):
```
[security]
allow_embedding = true
x_xss_protection = false

[auth]
disable_login_form = true
oauth_auto_login = true
oauth_allow_insecure_email_lookup = true

[auth.anonymous]
enabled = true
org_role = Viewer
hide_version = true

[auth.generic_oauth]
name = Azure AD
enabled = true
allow_sign_up = true
client_id = $OAUTH_CLIENT_ID
client_secret = $OAUTH_CLIENT_SECRET
scopes = openid email profile
auth_url = https://login.microsoftonline.com/xxxxxxxxxx/oauth2/v2.0/authorize
token_url = https://login.microsoftonline.com/xxxxxxxxxxxx/oauth2/v2.0/token
api_url: https://graph.microsoft.com/oidc/userinfo
role_attribute_path = contains(roles[*], 'Admin') && 'Admin' || contains(roles[*], 'Editor') && 'Editor' || 'Viewer
```

The scenario:
- i deleted sql lite db by pod restart
- the anonymous user visits a dashboard
  In ui i get the error message:
  You'll need additional permissions to perform this action. Permissions needed: dashboards:read
  
  In log i see:
  ```
logger=authn.service t=2023-10-25T12:41:28.439011751Z level=warn msg="Failed to authenticate request" client=auth.client.session error="user token not found"
  logger=authn.service t=2023-10-25T12:41:29.206471217Z level=warn msg="Failed to authenticate request" client=auth.client.session error="user token not found"
  logger=context userId=0 orgId=1 uname= t=2023-10-25T12:41:29.207484439Z level=info msg="Access denied" namespace=anonymous userID= accessErrorID=ACE9109877847 permissions="action:dashboards:read scopes:"
  logger=context userId=0 orgId=1 uname= t=2023-10-25T12:41:29.207536751Z level=info msg="Request Completed" method=GET path=/api/dashboards/uid/FXk5NbPWz status=403 remote_addr=10.170.74.225 time_ms=1 duration=1.476624ms size=165 referer="https://grafana-solace.sbb-cloud.net/d/FXk5NbPWz/solace-vpn?kiosk=&orgId=1&theme=light&var-instance=tms%2Fcapacityplanner%2Fcaparoute%2Fd-local" handler=/api/dashboards/uid/:uid
  logger=authn.service t=2023-10-25T12:41:29.215076268Z level=warn msg="Failed to authenticate request" client=auth.client.session error="user token not found"
  logger=context userId=0 orgId=1 uname= t=2023-10-25T12:41:29.216072726Z level=info msg="Request Completed" method=GET path=/api/live/ws status=-1 remote_addr=10.170.74.225 time_ms=1 duration=1.325134ms size=0 referer= handler=/api/live/ws
  logger=authn.service t=2023-10-25T12:41:30.329330519Z level=warn msg="Failed to authenticate request" client=auth.client.session error="user token not found"
  logger=context userId=0 orgId=1 uname= t=2023-10-25T12:41:30.330469345Z level=info msg="Request Completed" method=GET path=/api/live/ws status=-1 remote_addr=10.170.74.225 time_ms=1 duration=1.664819ms size=0 referer= handler=/api/live/ws
  logger=authn.service t=2023-10-25T12:41:34.109639606Z level=warn msg="Failed to authenticate request" client=auth.client.session error="user token not found"
  logger=context userId=0 orgId=1 uname= t=2023-10-25T12:41:34.110772108Z level=info msg="Request Completed" method=GET path=/api/live/ws status=-1 remote_addr=10.190.72.128 time_ms=1 duration=1.594219ms size=0 referer= handler=/api/live/ws
  logger=authn.service t=2023-10-25T12:41:39.62583126Z level=warn msg="Failed to authenticate request" client=auth.client.session error="user token not found"
 ```
- A admin user loges in and visits the dashboard
- The next anonymous user is able to vie the dashboard

**Comment by lukaszgryglicki** [comment]
Would love to hear somebody telling me what tables do I need to make inserts/deletes or updates so I can add this as a pod initialization, I must use 8.X due to this, none of 9.X or 10.X allows user anonymous access anymore.

**Comment by GreenRover** [comment]
Please give this query a try:
INSERT OR IGNORE INTO permission (id, role_id, action, scope, created, updated) SELECT (id * 10000) + 1, 1, "dashboards:read", "dashboards:uid:" || uid, datetime(), datetime() from dashboard WHERE is_folder=0;

Because i use grafana as docker pod and an sqlLite db i can not test it.

**Comment by lukaszgryglicki** [comment]
I can try this on Friday the earliest.

**Comment by pavel-mokrushin** [comment]
I had the same problem with the rights to dashboards for unauthorized users after upgrading from version 9.2.20 to 10.2.0.
I noticed that all rights except Admin disappeared from the dashboards folders.
I use Grafana in docker, dashboards are provisioned.
![image](https://github.com/grafana/grafana/assets/82204499/b3e5db3b-a297-46a3-a46c-07b6fad1c30f)

**Comment by wilfriedroset** [comment]
I've encountered the same issue while upgrading from 10.1.5 to 10.2.0 community edition
With errors like:
* You'll need additional permissions to perform this action. Permissions needed: orgs:read
* You'll need additional permissions to perform this action. Permissions needed: org.users:read

Here is what I've tried which does **not fix the issue**:
* [configure RBAC](https://grafana.com/docs/grafana/latest/administration/roles-and-permissions/access-control/configure-rbac/) switch off `permission_validation_enabled` which is documented as `default=false` but in fact it is `true`
* [configure RBAC](https://grafana.com/docs/grafana/latest/administration/roles-and-permissions/access-control/configure-rbac/) switch on `reset_basic_roles`
* [wipe clean the permission](https://grafana.com/docs/grafana/latest/administration/roles-and-permissions/access-control/troubleshooting/#missing-dashboard-folder-or-data-source-permissions) similar to the SQL query above

I've rollbacked to 10.1.5 until I find a suitable fix.

**Comment by lukaszgryglicki** [comment]
The approach is not working:
```
INSERT OR IGNORE INTO permission (id, role_id, action, scope, created, updated) SELECT (id * 10000) + 1, 1, "dashboards:read", "dashboards:uid:" || uid, datetime(), datetime() from dashboard WHERE is_folder=0;
```

Permissions after this look like:
```
select * from permission;
id,role_id,action,scope,created,updated,kind,attribute,identifier
10001,1,dashboards:read,dashboards:uid:1001,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
20001,1,dashboards:read,dashboards:uid:1,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
30001,1,dashboards:read,dashboards:uid:1002,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
40001,1,dashboards:read,dashboards:uid:2,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
50001,1,dashboards:read,dashboards:uid:1003,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
60001,1,dashboards:read,dashboards:uid:7,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
70001,1,dashboards:read,dashboards:uid:1004,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
80001,1,dashboards:read,dashboards:uid:1005,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
90001,1,dashboards:read,dashboards:uid:5,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
100001,1,dashboards:read,dashboards:uid:56,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
110001,1,dashboards:read,dashboards:uid:55,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
120001,1,dashboards:read,dashboards:uid:4,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
130001,1,dashboards:read,dashboards:uid:74,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
140001,1,dashboards:read,dashboards:uid:64,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
150001,1,dashboards:read,dashboards:uid:50,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
160001,1,dashboards:read,dashboards:uid:8,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
170001,1,dashboards:read,dashboards:uid:66,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
180001,1,dashboards:read,dashboards:uid:9,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
190001,1,dashboards:read,dashboards:uid:61,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
200001,1,dashboards:read,dashboards:uid:47,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
210001,1,dashboards:read,dashboards:uid:62,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
220001,1,dashboards:read,dashboards:uid:63,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
230001,1,dashboards:read,dashboards:uid:11,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
240001,1,dashboards:read,dashboards:uid:1011,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
250001,1,dashboards:read,dashboards:uid:12,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
260001,1,dashboards:read,dashboards:uid:82,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
270001,1,dashboards:read,dashboards:uid:67,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
280001,1,dashboards:read,dashboards:uid:13,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
290001,1,dashboards:read,dashboards:uid:14,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
300001,1,dashboards:read,dashboards:uid:52,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
310001,1,dashboards:read,dashboards:uid:15,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
320001,1,dashboards:read,dashboards:uid:19,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
330001,1,dashboards:read,dashboards:uid:16,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
340001,1,dashboards:read,dashboards:uid:18,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
350001,1,dashboards:read,dashboards:uid:17,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
360001,1,dashboards:read,dashboards:uid:29,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
370001,1,dashboards:read,dashboards:uid:10,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
380001,1,dashboards:read,dashboards:uid:20,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
390001,1,dashboards:read,dashboards:uid:21,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
400001,1,dashboards:read,dashboards:uid:23,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
410001,1,dashboards:read,dashboards:uid:22,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
420001,1,dashboards:read,dashboards:uid:24,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
430001,1,dashboards:read,dashboards:uid:75,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
440001,1,dashboards:read,dashboards:uid:25,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
450001,1,dashboards:read,dashboards:uid:26,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
460001,1,dashboards:read,dashboards:uid:68,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
470001,1,dashboards:read,dashboards:uid:3,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
480001,1,dashboards:read,dashboards:uid:27,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
490001,1,dashboards:read,dashboards:uid:28,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
500001,1,dashboards:read,dashboards:uid:48,"2023-11-22 08:24:01","2023-11-22 08:24:01","","",""
```

Still getting:
<img width="729" alt="Zrzut ekranu 2023-11-22 o 09 20 12" src="https://github.com/grafana/grafana/assets/2469783/ca5d4f42-994a-4dc5-a75c-1dd547715c16">

**Comment by Macilias** [comment]
btw. I had the same bug with grafana 10.2 and the solution was to get back to grafana 10.1.5

**Comment by lukaszgryglicki** [comment]
In my case, the problem is that I cannot use an `anonymous` user AT ALL with ANY Grafana > 8.X (so 9.X, 10.X). I must log in to display any dashboard - this makes no sense, as I think we should be able to use Grafana for pages that don't require any auth (my DB connection is read-only, so no need to do any permissions).

**Comment by GreenRover** [comment]
For me the issue is fixed since 10.2.1 but all dashboards needs to be imported again.

**Comment by lukaszgryglicki** [comment]
Created a feature request for this: https://github.com/grafana/grafana/issues/78639

**Comment by canaz** [comment]
Had the same problem and actually got rid of it by adding a new folder like discribed here:
https://community.grafana.com/t/dashboard-loading-fails-due-to-permissions-needed-folders-read/104427

**Comment by lfmullen** [comment]
Adding a dummy / empty folder with view permission to Viewers fixed this for me too (workaround) allowing me to lock down my actual folders and not expose them to non-users.
